src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -1,46 +1,177 @@
+import { RootState } from '../../../redux/store';
+import { setNotification } from '../../../redux/notificationSlice';
+import { BackgroundDiv, InputContainerBox } from '../../../components/styled/Wrappers';
+import { PageSpinner } from '../../../components/PageSpinner';
+import { CristinPerson, FlatCristinPerson } from '../../../types/user.types';
+import { updateCristinPerson } from '../../../api/userApi';
+import { useFetch } from '../../../utils/hooks/useFetch';
+import { UrlPathTemplate } from '../../../utils/urlPaths';
+import { getValueByKey } from '../../../utils/user-helpers';
+import { isErrorStatus, isSuccessStatus } from '../../../utils/constants';
 
 export const MyProfile = () => {
   const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const location = useLocation();
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const user = useSelector((store: RootState) => store.user)!; // If user has been empty this route would already be blocked
 
+  const currentCristinId = useSelector((store: RootState) => store.user?.cristinId) ?? '';
+  const isPublicPage = location.pathname === UrlPathTemplate.ResearchProfile;
+
+  const personId = isPublicPage
+    ? new URLSearchParams(location.search).get('id') ?? '' // Page for Research Profile of anyone
+    : currentCristinId; // Page for My Research Profile
+
+  const [person, isLoadingPerson, refetchPerson] = useFetch<CristinPerson>({
+    url: personId,
+    errorMessage: t('feedback.error.get_person'),
+  });
+
+  const firstName = getValueByKey('FirstName', person?.names);
+  const lastName = getValueByKey('LastName', person?.names);
+  const personPreferredFirstName = getValueByKey('PreferredFirstName', person?.names);
+  const personPreferredLastName = getValueByKey('PreferredLastName', person?.names);
+  const [editPreferredNames, setEditPreferredNames] = useState(false);
+
+  type CristinPersonFormData = Pick<FlatCristinPerson, 'preferredFirstName' | 'preferredLastName'>;
+
+  const initialValues: CristinPersonFormData = {
+    preferredFirstName: personPreferredFirstName || firstName,
+    preferredLastName: personPreferredLastName || lastName,
+  };
+
+  const updatePerson = async (values: CristinPersonFormData) => {
+    if (user.cristinId) {
+      if (values.preferredFirstName === '') {
+        values.preferredFirstName = null;
+      }
+      if (values.preferredLastName === '') {
+        values.preferredLastName = null;
+      }
+      const updatePersonResponse = await updateCristinPerson(user.cristinId, values);
+      if (isErrorStatus(updatePersonResponse.status)) {
+        dispatch(setNotification({ message: t('feedback.error.update_person'), variant: 'error' }));
+      } else if (isSuccessStatus(updatePersonResponse.status)) {
+        dispatch(setNotification({ message: t('feedback.success.update_person'), variant: 'success' }));
+        setEditPreferredNames(!editPreferredNames);
```

Comment:
Ingen biggie, men regner med vi alltid ønsker at preferred name skal være disabled etter lagring, så da er det greit å være eksplisitt på det
```suggestion
        setEditPreferredNames(false);
```